js/layout-javascript/agenda-code.js

```diff
@@ -909,16 +909,71 @@ DynamicList.prototype.attachObservers = function() {
                   return _this.deleteEntry(entryID);
                 })
                 .then(function onRemove(entryId) {
-                  _.remove(_this.listItems, function(entry) {
+                  var removedEntry = _.first(_.remove(_this.listItems, function(entry) {
                     return entry.id === parseInt(entryId, 10);
-                  });
+                  }));
 
                   _that.text(T('widgets.list.dynamic.notifications.confirmDelete.action')).removeClass('disabled');
                   _this.closeDetails({ focusOnEntry: event.type === 'keydown' });
 
                   _this.removeListItemHTML({
                     id: entryId
                   });
+
+                  // Make agenda layout empty if no entry left
+                  if (!_this.listItems.length) {
+                    _this.$container.find('.agenda-date-selector').addClass('hidden');
+                    _this.$container.find('.agenda-list-holder').addClass('hidden');
+
+                    // Toggle between filtered list and search view
+                    _this.toggleListView();
+
+                    _this.$container.find('.new-agenda-list-container').toggleClass('no-results', !_this.listItems.length);
+
+                    return;
+                  }
+
+                  // Delete list item from agendasByDay as well
+                  var foundDateField = _.find(_this.data.detailViewOptions, { location: _this.dateFieldLocation });
+                  var dateField = _.get(foundDateField, 'column');
+                  var agendasDayIndex = _this.getDateIndex(removedEntry.data[dateField]);
+                  var agendaDay = _this.agendasByDay[agendasDayIndex];
+
+                  _.remove(agendaDay, function(entry) {
+                    return entry.id === parseInt(entryId, 10);
+                  });
+
+                  // Do not change date if selected date has agendas to display
+                  if (agendaDay.length) {
+                    return;
+                  }
+
+                  // Update the date as selected date left with no agendas
+                  var indexToBeSet =  _this.$container
+                    .find('.agenda-date-selector li')
+                    .not('.placeholder')
+                    .index(_this.$container.find('.agenda-date-selector li.active'));
+                  var isNextIndexNotAvailable = _this.$container.find('.agenda-date-selector li.active').next().hasClass('placeholder');
```

Comment:
This will reduce unnecessary negation in the naming and logic.

```suggestion
                  var isNextIndexAvailable = !_this.$container.find('.agenda-date-selector li.active').next().hasClass('placeholder');
```